mevislab.github.io/content/tutorials/thirdparty/matplotlib/modulesetup.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Example 1: Setting up the module"
+date: 2023-05-26
+status: "OK"
+draft: false
+weight: 881
+tags: ["Advanced", "Tutorial", "Matplotlib", "Visualization"]
+menu: 
+  main:
+    identifier: "matplotlibexample1"
+    title: "Example 1: Setting up the module"
+    weight: 881
+    parent: "matplotlib"
+---
+
+# Example 1: Setting up the module
+
+To be able to acess the data needed for our grayscale distribution plots, we need a network consisting of a module that imports a DICOM, a module that differentiates between slices and another that ouputs histogram data. 
+
+So open up your MeVisLab workspace and add the modules `LocalImage`, `SubImage` and `Histogram` to it.
+Connect the output of `LocalImage` to the input of `SubImage` and the output of `SubImage` with the input of `Histogram`.
+If you feel like using a shortcut, you can also download the base network below and open it in your MeVisLab.
+
+Your finished network should look like this: 
+
+![Base network](/images/tutorials/thirdparty/Matplotlib1.PNG)
+
+{{< networkfile "/tutorials/thirdparty/matplotlib/MatplotlibBaseNetwork.mlab" >}}
+
+### Excursion on the concept behind modules
+
+To be able to build on the foundation we just set, it can be useful to understand how modules are conceptualized:
+You will have noticed how, for every module, a panel will pop up if you double-click it. The modules panel contains all of its functional parameters and enables you, as the user, to change them within a graphical user interface (GUI). We will do something similar later on. 
+But where and how is a module panel created? To answer this question, please close the module panel and right-click on the module. 
+A context menu will open, click on "Related Files". 
+
+![Context menu of the "SubImage" module](/images/tutorials/thirdparty/Matplotlib2.png)
+
+As you can see, each module has a .script and a .py file, named like the module itself: 
++ The .script file is, where the appearance and structure of the module panel as well as their commands are declared. 
++ The .py file contains Python functions and methods, whose are triggered by their referenced commands within the .script file.
+
+Some modules also reference a .mlab file which usually contains their internal network as the module is a macro. 
+
+**Let's continue with our module setup now:**  
+If your network is ready, group it by right-clicking on your group's title and select "Grouping", then "Add To A New Group".
+After, convert your grouped network into a macro module. 
+
+![Converting to a macro](/images/tutorials/thirdparty/Matplotlib3.png)
+{{<alert class="info" caption="Info">}}
+Information on how to convert groups into macros can be found [here](/tutorials/basicmechanisms#TutorialMacroModules). 
+{{</alert>}}
+Depending on whether or whether not you like to reuse your projects in other workspaces, it can make sense to convert to a global macro.
```

Comment:
whether or whether not -> wheater or not _or just_ wheater
convert -> convert them

Reply:
Abbreviated that as you requested